storage_service/locations/signals.py

```diff
@@ -142,3 +143,67 @@ def increment_model_save_count(sender, **kwargs):
     @receiver(signals.post_delete)
     def increment_model_delete_count(sender, **kwargs):
         model_delete_count.labels(model=sender.__name__).inc()
+
+
+def _user_is_administrator(cas_attributes):
+    """Determine if new user is an administrator from CAS attributes.
+
+    :param cas_attributes: Attributes dict returned by CAS server.
+
+    :returns: True if expected value is found, otherwise False.
+    """
+    ADMIN_ATTRIBUTE = settings.CAS_ADMIN_ATTRIBUTE
+    ADMIN_ATTRIBUTE_VALUE = settings.CAS_ADMIN_ATTRIBUTE_VALUE
+    if (ADMIN_ATTRIBUTE is None) or (ADMIN_ATTRIBUTE_VALUE is None):
+        LOGGER.error(
+            "Error determining if new user is an administrator. Please "
+            "be sure that env variables AUTH_CAS_ADMIN_ATTRIBUTE and "
+            "AUTH_CAS_ADMIN_ATTRIBUTE_VALUE are properly set."
+        )
+        return False
+
+    # CAS attributes are a dictionary. The value for a given key can be
+    # a string or a list, so our approach for checking for the expected
+    # value takes that into account.
+    ATTRIBUTE_TO_CHECK = cas_attributes.get(ADMIN_ATTRIBUTE)
+    if isinstance(ATTRIBUTE_TO_CHECK, list):
+        if ADMIN_ATTRIBUTE_VALUE in ATTRIBUTE_TO_CHECK:
+            return True
+    elif isinstance(ATTRIBUTE_TO_CHECK, str):
+        if ATTRIBUTE_TO_CHECK == ADMIN_ATTRIBUTE_VALUE:
+            return True
+    return False
+
+
+@receiver(cas_user_authenticated)
+def cas_user_authenticated_callback(sender, **kwargs):
```

Comment:
Should this handler belong to the `storage_service` app?

Reply:
Yes, I think you're right. And the receiver does not have the problem noted in the PR description when it's placed in `storage_service`. I've made the change in commit 144823d6153d42897fc891dec19a46026528c53b.